h/views/badge.py

```diff
@@ -22,6 +24,17 @@ def record_metrics(count,
     record_metric('Custom/Badge/badgeCountIsZero', int(count == 0))
 
 
+def _has_uri_ever_been_annotated(db, uri):
+    """Return `True` if a given URI has ever been annotated."""
+
+    # This check is written with SQL directly to guarantee an efficient
+    # query and minimize SQLAlchemy overhead.
```

Comment:
This is smart. Why spin up the whole ORM to ask whether something exists?

99% of the time, the developer efficiency we gain from the ORM is worth the overhead, but I think this is one of those 1% of times when raw SQL makes sense.

Reply:
Yeah, this is a great idea. When trying to optimize one particular code path that gets hit a lot like this, going down to SQL directly makes sense (assuming the efficiency gain does turn out to be significant)